src/main/java/com/stripe/model/BalanceTransaction.java

```diff
@@ -146,7 +148,16 @@ public String getType() {
 	public void setType(String type) {
 		this.type = type;
 	}
+	
+	public void setSourceObject(HasId sourceObject) {
+		this.sourceObject = sourceObject;
+	}
 
+	@SuppressWarnings("unchecked")
+	public <O> O getSourceObject() {
+		return (O) sourceObject;
+	}
```

Comment:
Sorry the delayed reply here @kion! I'm doing a horrible job getting this through.

> The problem with it was that it's not compatible with webhooks b/c data structure that is submitted to webhook handler doesn't support expandable fields and it was failing b/c of that.

If you understand why this is, could you elaborate a little more?

I saw the error on your other PR, but it seems to me that this should be working. Basically the `ExpandableField` works [by checking whether a value is a primitive or an object](https://github.com/stripe/stripe-java/blob/0c6c358a670af7967edfe60a9441c67b5495d7da/src/main/java/com/stripe/model/ExpandableFieldDeserializer.java#L17,L39) and should be functional even for data that comes in via webhook. I think we should see if we can get whatever problem this is fixed before falling back to a different solution.